src/bin/cargo-flamegraph.rs

```diff
@@ -306,11 +306,47 @@ fn workload(
     result
 }
 
+fn find_unique_bin_target() -> String {
+    let mut metadata_command = MetadataCommand::new();
+    metadata_command.no_deps();
+    let metadata = metadata_command
+        .exec()
+        .expect("failed to access crate metadata");
+
+    let bin_targets: Vec<String> = metadata
+        .packages
+        .into_iter()
+        .flat_map(|p| p.targets.into_iter())
+        .filter(|t| t.kind.contains(&"bin".into()))
+        .map(|t| t.name)
```

Comment:
We should use `filter_map()` instead of `filter().map()`. Ideally we should combine it with `flat_map()`, too, if at all possible.

Reply:
I combined it to `filter_map` but I didn't see a way to combine it with `flat_map`.